### PR TITLE
Added feature level test coverage to Alias provider (#2412)

### PR DESCRIPTION
**Pull request by ChrisUbben**
Adding more coverage to issue #2412
- Added more cmdlet test coverage for Alias provider.
- Moved all basic cmdlet test into a context block. (this is why the diff looks so messy)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1 [hunk 1]
```diff
@@ -1,55 +1,112 @@
...
-        $aliasExists | Should Be $true
+        It "Test executing the new alias" {
+            $result = Invoke-Expression $testAliasName
+            $result.GetType().Name | Should Be "DateTime"
+        }
     }
+}
+
+Describe "Extended Alias Provider Tests" -Tags "Feature" {
+    Context "Valdiation of Set-Item parameters for the Alias Provider" {
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1]: Context is not necessary if there is only one Context within the Describe.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1 [hunk 2]
```diff
@@ -1,55 +1,112 @@
-Describe "Basic Validation of Alias Provider" -Tags "CI" {
-    BeforeAll {
-        $testAliasName = 'TestAlias'
-        $testAliasValue = 'Get-Date'
-    }
+Describe "Basic Alias Provider Tests" -Tags "CI" {
+    Context "Validate basic Alias Cmdlets" {
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1]: Context is not necessary here

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1 (hunk 2 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1]: I made this change as a stylistic choice to be consistent with all of the other provider tests which do have more than 1 context block within the "Basic Alias Provider Tests" describe.
If you still think that I should remove the Context block please let me know.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1 (hunk 1 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1]: I plan to add more cmdlet coverage to the "Extended Alias Provider Tests" describe block in the future. I was preparing for that future by putting the context blocks in now.
If you still think that I should remove the Context block please let me know.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1 (hunk 2 above)
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1]: I'd like you to remove it here. It is unnecessary for the test.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1 (hunk 1 above)
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1]: In this case it seems OK to leave it.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1 (hunk 2 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Alias.Tests.ps1]: Fixed.

**Review by mirichmo — APPROVED** [review]

**Pull request merged**